.github/workflows/federated-integ-test.yml

```diff
@@ -13,14 +13,10 @@ name: Federated integ test
 
 on:  # yamllint disable-line rule:truthy
   workflow_dispatch: null
-  workflow_run:
-    workflows:
-      - build-all
+  push:
```

Comment:
For that workflow I agree, that there does not seem to be an obvious dependency on the build-all workflow. However, we need to decouple a bit more thoroughly.
- `github.event.workflow_run` dependencies need to be removed.
- The `federated-integ-test` job need to get a dependency on the `docker-build-*` jobs above.

Reply:
- I don't think it matters if they are set to push, but I can replace it with `github.event_name == 'push'`.
- Why would the second point be necessary? I am not aware of changing this logic here.

Reply:
* If you remove the `workflow_run` trigger those parts can probably be completely removed without replacement. Better not to leave or introduce dead code.
* Maybe not, but the logic is broken here anyway. And if we do an update to workflow triggers, we should probably leave the workflow in a "not logically broken" state afterwards.
A `needs: [docker-build-orc8r, docker-build-feg]` is required for the integration test workflow, as you actually want to test the changes for the current commit there, which must previously be build.

Reply:
If we need to update the logic of the dependencies within this job then I would prefer to do it in a separate PR since this is completely unrelated to the overall workflow triggers.

Reply:
https://github.com/magma/magma/pull/14239